arwen/elrondapi/managedei.go

```diff
@@ -926,3 +957,102 @@ func v1_4_managedTransferValueExecute(
+	metering := arwen.GetMeteringContext(context)
+	blockchain := arwen.GetBlockchainContext(context)
+	managedType := arwen.GetManagedTypesContext(context)
+
+	gasToUse := metering.GasSchedule().ElrondAPICost.GetExternalBalance
+	metering.UseGasAndAddTracedGas(managedIsESDTFrozenName, gasToUse)
+
+	address, err := managedType.GetBytes(addressHandle)
+	if err != nil {
+		_ = arwen.WithFault(arwen.ErrArgOutOfRange, context, runtime.ElrondAPIErrorShouldFailExecution())
+		return -1
+	}
+	tokenID, err := managedType.GetBytes(tokenIDHandle)
+	if err != nil {
+		_ = arwen.WithFault(arwen.ErrArgOutOfRange, context, runtime.ElrondAPIErrorShouldFailExecution())
+		return -1
+	}
+
+	esdtToken, err := blockchain.GetESDTToken(address, tokenID, uint64(nonce))
+	if err != nil {
+		_ = arwen.WithFault(arwen.ErrArgOutOfRange, context, runtime.ElrondAPIErrorShouldFailExecution())
+		return -1
+	}
+
+	esdtUserData := builtInFunctions.ESDTUserMetadataFromBytes(esdtToken.Properties)
+	if esdtUserData.Frozen {
+		return 1
+	}
+	return 0
+}
+
+//export v1_4_managedIsLimitedTransfer
+func v1_4_managedIsLimitedTransfer(context unsafe.Pointer, tokenIDHandle int32) int32 {
+	runtime := arwen.GetRuntimeContext(context)
+	metering := arwen.GetMeteringContext(context)
+	blockchain := arwen.GetBlockchainContext(context)
+	managedType := arwen.GetManagedTypesContext(context)
+
+	gasToUse := metering.GasSchedule().ElrondAPICost.GetExternalBalance
+	metering.UseGasAndAddTracedGas(managedIsLimitedTransferName, gasToUse)
+
+	tokenID, err := managedType.GetBytes(tokenIDHandle)
+	if err != nil {
+		_ = arwen.WithFault(arwen.ErrArgOutOfRange, context, runtime.ElrondAPIErrorShouldFailExecution())
+		return -1
+	}
+
+	if blockchain.IsLimitedTransfer(tokenID) {
+		return 1
+	}
+
+	return 0
+}
+
+//export v1_4_managedIsPaused
+func v1_4_managedIsPaused(context unsafe.Pointer, tokenIDHandle int32) int32 {
+	runtime := arwen.GetRuntimeContext(context)
+	metering := arwen.GetMeteringContext(context)
+	blockchain := arwen.GetBlockchainContext(context)
+	managedType := arwen.GetManagedTypesContext(context)
+
+	gasToUse := metering.GasSchedule().ElrondAPICost.GetExternalBalance
+	metering.UseGasAndAddTracedGas(managedIsPausedName, gasToUse)
+
+	tokenID, err := managedType.GetBytes(tokenIDHandle)
+	if err != nil {
+		_ = arwen.WithFault(arwen.ErrArgOutOfRange, context, runtime.ElrondAPIErrorShouldFailExecution())
+		return -1
+	}
+
+	if blockchain.IsPaused(tokenID) {
+		return 1
+	}
+
+	return 0
+}
+
+//export v1_4_managedBufferToHex
+func v1_4_managedBufferToHex(context unsafe.Pointer, sourceHandle int32, destHandle int32) {
+	runtime := arwen.GetRuntimeContext(context)
+	metering := arwen.GetMeteringContext(context)
+	managedType := arwen.GetManagedTypesContext(context)
+
+	gasToUse := metering.GasSchedule().ManagedBufferAPICost.MBufferSetBytes
+	metering.UseGasAndAddTracedGas(managedBufferToHexName, gasToUse)
+
+	mBuff, err := managedType.GetBytes(sourceHandle)
```

Comment:
In this case, shouldn't we perhaps use a little bit of gas with respect to the size of the buffer, as well?

Reply:
it is fine - as gas was taken out once when the sourceHandle was created.